src/app/zap-templates/templates/chip/helper.js

```diff
@@ -470,6 +470,27 @@ async function if_chip_enum(type, options)
   return templateUtil.templatePromise(this.global, result);
 }
 
+async function if_chip_complex(options)
+{
+  if (this.isArray) {
+    return options.fn(this);
+  }
+
+  if (this.type == 'array') {
+    return options.fn(this);
+  }
+
+  let pkgId       = await templateUtil.ensureZclPackageId(this);
+  let checkResult = await zclHelper.isStruct(this.global.db, this.type, pkgId);
+  let result;
+  if (checkResult != 'unknown') {
```

Comment:
why does isStruct return a string 'unknown'? Based on the naming it seems it should return a true/false (isStruct or not isStruct).

Reply:
I agree that the naming is confusing. I have opened https://github.com/project-chip/zap/issues/413 for it.